lib/jekyll/theme_builder.rb

```diff
@@ -0,0 +1,108 @@
+class Jekyll::ThemeBuilder
+  SCAFFOLD_DIRECTORIES = %w{
+    _layouts _includes _sass
+  }.freeze
+
+  attr_reader :name, :path
+
+  def initialize(theme_name)
+    @name = theme_name.to_s.gsub(/ /, "_").gsub(/_+/, "_")
+    @path = Pathname.new(File.expand_path(name, Dir.pwd))
+  end
+
+  def create!
+    create_directories
+    create_gemspec
+    create_accessories
+    initialize_git_repo
+  end
+
+  private
+
+  def root
+    @root ||= Pathname.new(File.expand_path("../", __dir__))
+  end
+
+  def template_file(filename)
+    [
+      root.join("theme_template", "#{filename}.erb"),
+      root.join("theme_template", "#{filename}")
+    ].find do |pathname|
+      pathname.exist?
+    end
+  end
+
+  def template(filename)
+    erb.render(template_file(filename).read)
+  end
+
+  def erb
+    @erb ||= ERBRenderer.new(self)
+  end
+
+  def mkdir_p(directories)
+    Array(directories).each do |directory|
+      full_path = path.join(directory)
+      Jekyll.logger.info "create", "#{full_path}"
+      FileUtils.mkdir_p(full_path)
+    end
+  end
+
+  def write_file(filename, contents)
+    full_path = path.join(filename)
+    Jekyll.logger.info "create", "#{full_path}"
+    File.write(full_path, contents)
```

Comment:
This is a good candidate for `Pathutil#write` since it's Windows compatible and writes Windows compatible line endings on your behalf.  This way Windows users get what they use natively and Linux/Unix will too.